nifi-nar-bundles/nifi-grpc-bundle/nifi-grpc-common/src/main/java/org/apache/nifi/processors/grpc/FlowFileIngestServiceInterceptor.java

```diff
@@ -100,10 +104,11 @@ public <I, O> ServerCall.Listener<I> interceptCall(
         final SSLSession sslSession = attributes.get(Grpc.TRANSPORT_ATTR_SSL_SESSION);
         if (this.authorizedDNPattern != null && sslSession != null) {
             try {
-                final X509Certificate[] certs = sslSession.getPeerCertificateChain();
+                final Certificate[] certs = sslSession.getPeerCertificates();
                 if (certs != null && certs.length > 0) {
-                    for (final X509Certificate cert : certs) {
-                        foundSubject = cert.getSubjectDN().getName();
+                    for (final Certificate cert : certs) {
+                        final X509Certificate x509Cert = CertificateUtils.convertAbstractX509Certificate(cert);
```

Comment:
This convert method can be replaced with a simple cast, avoiding the dependency on `nifi-security-utils`.
```suggestion
                        final X509Certificate x509Cert = (X509Certificate) cert;
```

A private method could be introduced to run an `instanceof` check on the `Certificate` object, but the end result would still be some subclass of a `RuntimeException`.